### DISCUSSION TITLE
issue: trigger with shouldFocus true causing page unresponsive when clicking on Select

### DISCUSSION THREAD

**Discussion by DominicGBauer** [react-hook-form/react-hook-form#9487] [Q&A] [answered]
### Version Number

7

### Codesandbox/Expo snack

https://codesandbox.io/s/jovial-moore-n69k7w

### Steps to reproduce

There seems to be an issue with onFocus in select component causing an issue when trigger with the shouldFocus option is used on another component resulting in the page becoming unresponsive

1. Go to 'https://codesandbox.io/s/jovial-moore-n69k7w'
2. Click directly on "Tier" select (before any validation as run)
3. It should make the page unresponsive

### Expected behaviour

I should be able to allow trigger with shouldFocus true even if my form has non text fields

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@DominicGBauer 

This is not a bug. Your field `firstName` is set `autofocus` so it will be focused by default by browser but its `onBlur` callback also triggers validation and focus on error when end-user leaves `firstName` field blank to select on `tier` field. 

This handling will cause losing focus on `firstName` and focus on `tier` but then validation is executed and set focus again on error of `firstName`.

Just remove `onBlur` from `firstName` and trigger validation for `firstName` in `onChange` callback of `tier` if you want to validate `firstName` at that time.
